src/main/fc/core.c

```diff
@@ -996,10 +996,10 @@ void processRxModes(timeUs_t currentTimeUs)
     if (FLIGHT_MODE(ANGLE_MODE) || FLIGHT_MODE(HORIZON_MODE)) {
         LED1_ON;
         // increase frequency of attitude task to reduce drift when in angle or horizon mode
-        rescheduleTask(TASK_ATTITUDE, TASK_PERIOD_HZ(500));
+        rescheduleTask(TASK_ATTITUDE, TASK_PERIOD_HZ(lrintf(acc.sampleRateHz / (float)imuConfig()->imu_process_denom)));
```

Comment:
(you can probably skip `lrintf` here, TASK_PERIOD_HZ will convert it to microseconds. Or calculate microseconds directly (imuConfig()->imu_process_denom * 1000000 / acc.sampleRateHz) )

Reply:
You are right. I will fix it.